runtime/src/main/java/org/corfudb/protocols/wireprotocol/IMetadata.java

```diff
@@ -206,6 +207,26 @@ default void setCheckpointedStreamStartLogAddress(Long startLogAddress) {
         getMetadataMap().put(CHECKPOINTED_STREAM_START_LOG_ADDRESS, startLogAddress);
     }
 
+    default void setPayloadCodecType(Codec.Type type) {
+        getMetadataMap().put(LogUnitMetadataType.PAYLOAD_CODEC, type);
+    }
+
+    default Codec.Type getPayloadCodecType() {
+        return (Codec.Type) getMetadataMap().getOrDefault(LogUnitMetadataType.PAYLOAD_CODEC, Codec.Type.NONE);
+    }
+
+    default boolean hasPayloadCodec() {
+        return getPayloadCodecType() != Codec.Type.NONE;
+    }
+
+    default boolean isCompressed() {
+        return (boolean) getMetadataMap().getOrDefault(LogUnitMetadataType.COMPRESSED, false);
+    }
```

Comment:
I am a bit confused -- why do you need both `hasPayloadCodec` and `isCompressed`? Seem like if the entry is not compressed it will always have its codec set to `NONE`.

Reply:
In the code (LogData.java L. 274) there is a line that states the following:

"Since the server uses the same serialization/deserialization path, we need a flag to let the server not re-encode an already encoded payload)"

I'll add this comment in this point of the code so it serves as reference.